lib/qiita/markdown/filters/inline_code_color.rb

```diff
@@ -0,0 +1,37 @@
+module Qiita
+  module Markdown
+    module Filters
+      class InlineCodeColor < HTML::Pipeline::Filter
+        REGEXPS = Regexp.union(
+          /\#([0-9a-fA-F]{3}|[0-9a-fA-F]{6})/,
+          /rgba?\(\s*(\d+(\,|\s)\s*){2}\d+\s*\)/,
+          /rgba?\(\s*(\d+%(\,|\s)\s*){2}\d+%\s*\)/,
+          /rgba?\(\s*(\d+\,\s*){3}\d*\.?\d+%?\s*\)/,
+          /rgba?\(\s*(\d+\s*){2}\d+\s*\/\s*\d?\.?\d+%?\s*\)/,
+          /rgba?\(\s*(\d+%\s*){2}\d+%\s*\/\s*\d?\.?\d+%?\s*\)/,
+          /hsla?\(\s*\d+(deg|rad|grad|turn)?\,\s*\d+%\,\s*\d+%\s*\)/,
+          /hsla?\(\s*\d+(deg|rad|grad|turn)?\s+\d+%\s+\d+%\s*\)/,
+          /hsla?\(\s*\d+(deg|rad|grad|turn)?\,\s*(\d+%\,\s*){2}\d?\.?\d+%?\s*\)/,
+          /hsla?\(\s*\d+(deg|rad|grad|turn)?\s+\d+%\s+\d+%\s*\/\s*\d?\.?\d+%?\s*\)/,
+        )
+
+        COLOR_CODE_PATTERN = /\A\s*(#{REGEXPS})\s*\z/
+
+        def call
+          doc.search(".//code").each do |node|
+            if (color = node.inner_text) =~ COLOR_CODE_PATTERN
+              node.add_child(color_element(color))
+            end
+          end
+          doc
+        end
+
+        private
+
+        def color_element(color)
+          %(<span class="" style="background-color: #{color};"></span>)
```

Comment:
Please fill the class attribute. I think it's worth defining a context (such as `:inline_code_class_name`) to provide a way to override the default class value.